[Bug]: 32-bit build fails on mcp_client.rs due to leftover AtomicU64 (missed in PR #3409)

### Affected component

unknown

### Severity

S1 - workflow blocked

### Current behavior

The recent merge from PR #3409 successfully fixed the 64-bit atomic counters across the `observability` and `channels` modules for 32-bit architectures. However, the `src/tools/mcp_client.rs` file was missed by the AI agent during that PR. 

It still explicitly imports `std::sync::atomic::AtomicU64`, which causes an `E0432` unresolved import error on 32-bit targets. If `AtomicU64` is manually downgraded to `AtomicU32` locally to bypass that, the compiler then throws an `E0308` type mismatch error on line 160 because `JsonRpcRequest::new()` expects a standard `u64` ID, not a `u32`.

### Expected behavior

`src/tools/mcp_client.rs` should use the same conditional compilation (`cfg`) strategy for atomics as the rest of the patched codebase. It should fall back to `AtomicU32` on 32-bit targets, and properly cast the generated ID using `.into()` or `as u64` when passing it into the JSON-RPC request at line 160, allowing the agent to compile flawlessly on 32-bit architectures.

### Steps to reproduce

```bash
1. Add target and nightly components for strict embedded 32-bit testing:
   rustup target add mipsel-unknown-linux-musl
   rustup toolchain install nightly
   rustup component add rust-src --toolchain nightly
2. Run the build simulating a strict soft-float MIPS environment on the master branch:
   RUSTFLAGS="-C target-feature=+soft-float" cargo +nightly build --release --target mipsel-unknown-linux-musl --no-default-features -Z build-std=std,panic_abort
3. Observe the E0432 import crash in mcp_client.rs.
```

### Impact

Affected users: Anyone building or cross-compiling for 32-bit edge devices (e.g., MIPS routers, older ARM boards).
Frequency: Always (hard block on compilation).
Consequence: Prevents the master branch from compiling on 32-bit hardware without manual source-code edits.

### Logs / stack traces

```text
error[E0432]: unresolved import `std::sync::atomic::AtomicU64`
 --> src/tools/mcp_client.rs:6:25
  |
6 | use std::sync::atomic::{AtomicU64, Ordering};
  |                         ^^^^^^^^^ no `AtomicU64` in `sync::atomic`

// After manually changing to AtomicU32, the missing cast error appears:
error[E0308]: mismatched types
   --> src/tools/mcp_client.rs:160:13
    |
159 |         let req = JsonRpcRequest::new(
    |                   ------------------- arguments to this function are incorrect
160 |             id,
    |             ^^ expected `u64`, found `u32`
```

### ZeroClaw version

master (post PR #3409 merge)

### Rust version

nightly

### Operating system

Target: mipsel-unknown-linux-musl

### Regression?

Unknown

### Pre-flight checks

- [x] I reproduced this on the latest master branch or latest release.
- [x] I redacted secrets, tokens, and personal data from all submitted content.

--- a/src/tools/mcp_client.rs
+++ b/src/tools/mcp_client.rs
@@ -3,7 +3,11 @@
 //! Supports multiple transports: stdio (spawn local process), HTTP, and SSE.
 
 use std::collections::HashMap;
-use std::sync::atomic::{AtomicU64, Ordering};
+#[cfg(not(target_has_atomic = "64"))]
+use std::sync::atomic::AtomicU32;
+#[cfg(target_has_atomic = "64")]
+use std::sync::atomic::AtomicU64;
+use std::sync::atomic::Ordering;
 use std::sync::Arc;
 
 use anyhow::{anyhow, bail, Context, Result};
@@ -32,7 +36,10 @@ const MAX_TOOL_TIMEOUT_SECS: u64 = 600;
 struct McpServerInner {
     config: McpServerConfig,
     transport: Box<dyn McpTransportConn>,
+    #[cfg(target_has_atomic = "64")]
     next_id: AtomicU64,
+    #[cfg(not(target_has_atomic = "64"))]
+    next_id: AtomicU32,
     tools: Vec<McpToolDef>,
 }
 
@@ -123,7 +130,10 @@ impl McpServer {
         let inner = McpServerInner {
             config,
             transport,
+            #[cfg(target_has_atomic = "64")]
             next_id: AtomicU64::new(3), // Start at 3 since we used 1 and 2
+            #[cfg(not(target_has_atomic = "64"))]
+            next_id: AtomicU32::new(3), // Start at 3 since we used 1 and 2
             tools: tool_list.tools,
         };
 
@@ -155,7 +165,7 @@ impl McpServer {
         arguments: serde_json::Value,
     ) -> Result<serde_json::Value> {
         let mut inner = self.inner.lock().await;
-        let id = inner.next_id.fetch_add(1, Ordering::Relaxed);
+        let id = inner.next_id.fetch_add(1, Ordering::Relaxed) as u64;
         let req = JsonRpcRequest::new(
             id,
             "tools/call",

